lib/client.js

```diff
@@ -78,6 +81,11 @@ module.exports = function (dependencies) {
               this.config.token.regenerate(tokenGeneration);
               resolve(this.write(notification, device, retryCount + 1));
               return;
+            } else if (status === "500" && response.reason === "InternalServerError") {
+              stream.connection.close();
+              let error = new VError("Error 500, stream ended unexpectedly");
```

Comment:
Oops, I missed the **success** case. 

But in my understanding, the reason why it adds `error` in return object is that it would not receive response from server side and there would not status code. [this case](https://github.com/cmdkoh/node-apn/blob/710f68c9fb8ddd9cd71b089221ff61f7cb2a8fd0/lib/client.js#L92)

So, I don't understand why this library should add error when it receives specific response from server, eg: 500.

It would be very appreciated if you could explain your concerns to me. @jpwilliams